app/controllers/controllers.js

```diff
@@ -7,4 +7,23 @@ exports.getTopics = async (req, res, next) => {
   } catch (err) {
     next(err);
   }
-};
\ No newline at end of file
+};
+
+const { fetchArticleById } = require("../models/articles");
+
+exports.getArticleById = async (req, res, next) => {
+  try {
+    const { article_id } = req.params;
+    const article = await fetchArticleById(article_id);
+    if (!article) {
+      return res.status(404).send({ msg: "Article not found" });
+    }
+    res.status(200).send({ article });
+  } catch (err) {
+    if (err.code === "22P02") {
+      res.status(400).send({ msg: "Invalid article ID" });
+    } else {
+      next(err);
+    }
+  }
```

Comment:
To make things here less verbose, we want to handle our errors with an error handling middleware most of the time. You currently don't have one for a 500 Internal Server Error, and you could absolutely do with one for PSQL errors, and if you wanted to you could even have one for custom errors. Check out the notes - no need to clutter up our controllers like this! :-)

Reply:
I did think about this when I was writing it but still feel a bit unsure on how to use middleware 😅 I'll go over the notes before I move on to the next task and refactor my error handling, thanks for the feedback 🙏